tests/e2e/gc_block_volume_expansion.go

```diff
@@ -2447,6 +2456,37 @@ func checkPvcHasGivenStatusCondition(client clientset.Interface, namespace strin
 	return pvclaim, nil
 }
 
+//  waitforPvctoGetStatusCondition checks & waits till specified time out if the status condition in PVC matches with the one we want
+
+func waitforPvctoGetStatusCondition(client clientset.Interface, namespace string, pvcName string,
+	timeout time.Duration, condition v1.PersistentVolumeClaimConditionType) (*v1.PersistentVolumeClaim, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	var pvclaim *v1.PersistentVolumeClaim
+	var err error
+	waitErr := wait.PollImmediate(resizePollInterval, timeout, func() (bool, error) {
+		pvclaim, err = client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvcName, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		inProgressConditions := pvclaim.Status.Conditions
+		// If there are conditions on the PVC, it must be of
+		// FileSystemResizePending type.
+		if len(inProgressConditions) > 0 {
+			expectEqual(len(inProgressConditions), 1, fmt.Sprintf("PVC '%v' has more than one status conditions", pvcName))
+			if inProgressConditions[0].Type == condition {
+				return true, nil
+			}
+		} else {
+			return false, fmt.Errorf(
```

Comment:
if len(inProgressConditions) == 0 then what condition are you matching here
IMO this else block is not required

Reply:
Fixed